### PR TITLE
Ewdurbin package local dir as whl

### PR DESCRIPTION
**Pull request by stealthycoin**
Expands on #653 to implement packaging arbitrary directories on your local machine.

Closes #653

### CODE REVIEW

**Comment by jamesls** [comment]
Seems reasonable, but it looks like this failed the type checker.

File: chalice/deploy/packager.py [hunk 1]
```diff
@@ -700,6 +714,19 @@ def download_all_dependencies(self, requirements_filename, directory):
                 package_name = match.group(1)
                 raise NoSuchPackageError(str(package_name))
             raise PackageDownloadError(error)
+        stdout = out.decode()
+        match = re.search(self._LINK_IS_DIR_PATTERN, stdout)
+        if match:
+            wheel_package_path = str(match.group(1))
```
Comment (jamesls) [inline chalice/deploy/packager.py]: Is the `str()` needed?  The `stdout` you pass `re.search` is already decoded.

Comment (stealthycoin) [reply chalice/deploy/packager.py]: That was to make mypy happy.

Comment (stealthycoin) [reply chalice/deploy/packager.py]:
```
chalice/deploy/packager.py:729: error: Argument 1 to "build_wheel" of "PipRunner" has incompatible type "unicode"; expected "str"
```

We do the same thing a few lines above for the same reason: https://github.com/aws/chalice/blob/master/chalice/deploy/packager.py#L701

Comment (jamesls) [reply chalice/deploy/packager.py]: Ah that's right, the whole `AnyStr` thing.  Yeah, shipit then (don't forget to squash your commits).

**Pull request merged**